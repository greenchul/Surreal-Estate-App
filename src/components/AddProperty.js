import React, { useState } from "react";
import "../styles/AddProperty.css";

const AddProperty = () => {
  const initialState = {
    fields: {
      title: "",
      type: "Flat",
      bedrooms: 0,
      bathrooms: 0,
      price: 0,
      city: "Manchester",
    },
  };
  // eslint-disable-next-line no-unused-vars
  const [fields, setFields] = useState(initialState.fields);

  const handleAddProperty = (e) => {
    e.preventDefault();
  };
  const handleFieldChange = (e) => {
    setFields({ ...fields, [e.target.name]: e.target.value });
  };
  return (
    <div className="add-property">
      <p>Add Properties</p>
      <form onSubmit={handleAddProperty} className="add-property__form">
        <label htmlFor="title">
          Title
          <input
            id="title"
            name="title"
            value={fields.title}
            onChange={handleFieldChange}
            placeholder="House for sale"
            className="add-property__form__field"
          />
        </label>

        <label htmlFor="type">
          Type
          <select
            id="type"
            name="type"
            value={fields.type}
            onChange={handleFieldChange}
            className="add-property__form__field"
          >
            <option value="Flat">Flat</option>
            <option value="Detached">Detached</option>
            <option value="Semi-Detached">Semi-Detached</option>
            <option value="Terraced">Terraced</option>
            <option value="End of Terrace">End of Terrace</option>
            <option value="Cottage">Cottage</option>
            <option value="Bungalow">Bungalow</option>
          </select>
        </label>

        <label htmlFor="bedrooms">
          Bedrooms
          <input
            type="number"
            id="bedrooms"
            placeholder="4"
            min="0"
            max="20"
            name="bedrooms"
            // value={fields.bedrooms}
            onChange={handleFieldChange}
            className="add-property__form__field"
          />
        </label>

        <label htmlFor="bathrooms">
          Bathroom
          <input
            type="number"
            id="bathrooms"
            min="0"
            max="15"
            name="bathrooms"
            // value={fields.bathrooms}
            onChange={handleFieldChange}
            className="add-property__form__field"
            placeholder="1"
          />
        </label>

        <label htmlFor="price">
          Price
          <input
            type="number"
            name="price"
            id="price"
            value={fields.price}
            min="0"
            onChange={handleFieldChange}
            className="add-property__form__field"
          />
        </label>

        <label htmlFor="city">
          City
          <select
            id="city"
            name="city"
            value={fields.city}
            onChange={handleFieldChange}
            className="add-property__form__field"
          >
            <option value="Manchester">Manchester</option>
            <option value="Leeds">Leeds</option>
            <option value="Sheffield">Sheffield</option>
            <option value="Liverpool">Liverpool</option>
          </select>
        </label>

        <label htmlFor="email">
          Email
          <input
            type="email"
            id="email"
            name="email"
            placeholder="hello@email.com"
            onChange={handleFieldChange}
            className="add-property__form__field"
          />
        </label>
        <div className="add-property__form__button">
          <button type="submit">Add</button>
        </div>
      </form>
    </div>
  );
};

export default AddProperty;
