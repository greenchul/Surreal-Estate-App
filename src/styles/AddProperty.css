.add-property {
  display: flex;
  flex-direction: column;
  align-items: center;
  font-size: 30px;
}
.add-property__form {
  padding: 20px;
  background-color: rgb(247, 245, 243);
  margin-top: 20px;
  width: 40%;
  display: flex;
  flex-direction: column;
  box-shadow: 0px 6px 1px rgba(67, 66, 66, 0.311);
  border-radius: 10px;
}

button {
  width: 50%;
  border-radius: 10px;
  margin-top: 20px;
  border: 1px solid black;
  background-color: rgb(153, 247, 188);
  cursor: pointer;
  padding: 10px;
  font-size: 20px;
  letter-spacing: 2px;
}

button:hover {
  background-color: rgb(122, 160, 136);
}

button:active {
  box-shadow: 0px 6px 1px rgba(67, 66, 66, 0.311);
  transform: translateY(4px);
}

label {
  display: flex;
  flex-direction: row;
  justify-content: space-between;
  margin-bottom: 10px;
}

input {
  width: 70%;
  border-radius: 10px;
  border: 1px solid black;
  text-indent: 10px;
  font-size: 20px;
}

select {
  width: 70%;
  border-radius: 10px;
  border: 1px solid black;
  text-indent: 10px;
  font-size: 20px;
}

.add-property__form__button {
  display: flex;
  justify-content: center;
}
