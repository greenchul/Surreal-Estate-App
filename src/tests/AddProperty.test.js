import React from "react";
import { render, screen } from "@testing-library/react";
import { BrowserRouter } from "react-router-dom";
import AddProperty from "../components/AddProperty";

describe("AddProperty component", () => {
  test("component renders correctly", () => {
    const { asFragment } = render(
      <BrowserRouter>
        <AddProperty />
      </BrowserRouter>
    );

    expect(asFragment()).toMatchSnapshot();
  });

  test("title input renders correctly", () => {
    render(
      <BrowserRouter>
        <AddProperty />
      </BrowserRouter>
    );
    const titleElement = screen.getByPlaceholderText("House for sale");

    expect(titleElement).toBeInTheDocument();
  });

  test("type input renders correctly", () => {
    render(
      <BrowserRouter>
        <AddProperty />
      </BrowserRouter>
    );
    const typeElement = screen.getByDisplayValue("Flat");

    expect(typeElement).toBeInTheDocument();
  });

  test("bedrooms input renders correctly", () => {
    render(
      <BrowserRouter>
        <AddProperty />
      </BrowserRouter>
    );
    const bedroomsElement = screen.getByPlaceholderText("4");

    expect(bedroomsElement).toBeInTheDocument();
  });

  test("bathrooms input renders correctly", () => {
    render(
      <BrowserRouter>
        <AddProperty />
      </BrowserRouter>
    );
    const bathroomsElement = screen.getByPlaceholderText("1");

    expect(bathroomsElement).toBeInTheDocument();
  });

  test("price input renders correctly", () => {
    render(
      <BrowserRouter>
        <AddProperty />
      </BrowserRouter>
    );
    const priceElement = screen.getByDisplayValue("0");

    expect(priceElement).toBeInTheDocument();
  });

  test("city input renders correctly", () => {
    render(
      <BrowserRouter>
        <AddProperty />
      </BrowserRouter>
    );
    const cityElement = screen.getByDisplayValue("Manchester");

    expect(cityElement).toBeInTheDocument();
  });

  test("email input renders correctly", () => {
    render(
      <BrowserRouter>
        <AddProperty />
      </BrowserRouter>
    );
    const emailElement = screen.getByPlaceholderText("hello@email.com");

    expect(emailElement).toBeInTheDocument();
  });
});
