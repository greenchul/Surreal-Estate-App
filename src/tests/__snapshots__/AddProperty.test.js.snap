// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddProperty component component renders correctly 1`] = `
<DocumentFragment>
  <div
    class="add-property"
  >
    <p>
      Add Properties
    </p>
    <form
      class="add-property__form"
    >
      <label
        for="title"
      >
        Title
        <input
          class="add-property__form__field"
          id="title"
          name="title"
          placeholder="House for sale"
          value=""
        />
      </label>
      <label
        for="type"
      >
        Type
        <select
          class="add-property__form__field"
          id="type"
          name="type"
        >
          <option
            value="Flat"
          >
            Flat
          </option>
          <option
            value="Detached"
          >
            Detached
          </option>
          <option
            value="Semi-Detached"
          >
            Semi-Detached
          </option>
          <option
            value="Terraced"
          >
            Terraced
          </option>
          <option
            value="End of Terrace"
          >
            End of Terrace
          </option>
          <option
            value="Cottage"
          >
            Cottage
          </option>
          <option
            value="Bungalow"
          >
            Bungalow
          </option>
        </select>
      </label>
      <label
        for="bedrooms"
      >
        Bedrooms
        <input
          class="add-property__form__field"
          id="bedrooms"
          max="20"
          min="0"
          name="bedrooms"
          placeholder="4"
          type="number"
        />
      </label>
      <label
        for="bathrooms"
      >
        Bathroom
        <input
          class="add-property__form__field"
          id="bathrooms"
          max="15"
          min="0"
          name="bathrooms"
          placeholder="1"
          type="number"
        />
      </label>
      <label
        for="price"
      >
        Price
        <input
          class="add-property__form__field"
          id="price"
          min="0"
          name="price"
          type="number"
          value="0"
        />
      </label>
      <label
        for="city"
      >
        City
        <select
          class="add-property__form__field"
          id="city"
          name="city"
        >
          <option
            value="Manchester"
          >
            Manchester
          </option>
          <option
            value="Leeds"
          >
            Leeds
          </option>
          <option
            value="Sheffield"
          >
            Sheffield
          </option>
          <option
            value="Liverpool"
          >
            Liverpool
          </option>
        </select>
      </label>
      <label
        for="email"
      >
        Email
        <input
          class="add-property__form__field"
          id="email"
          name="email"
          placeholder="hello@email.com"
          type="email"
        />
      </label>
      <div
        class="add-property__form__button"
      >
        <button
          type="submit"
        >
          Add
        </button>
      </div>
    </form>
  </div>
</DocumentFragment>
`;
